src/optionsdialog.cpp

```diff
@@ -179,6 +179,9 @@ void OptionsDialog::writeSettings(void)
     settings->endGroup();
     ///-----PROGRAMING--LANGUAGE---------------------------------------------------
     settings->beginGroup("compilation_settings");
+    settings->setValue("defaultLanguage/text",defaultLanguageComboBox->currentText());
```

Comment:
I really use only index, but in my task is said to save selected default language name. I thought , maybe, it would be necessary to use chosen program's name somewhere else in the program. Or is it unnecessary? Should I remove this piece of code?